### PR TITLE
These attributes are better for the final state(#13628)

### PR DESCRIPTION
**Pull request by mrhbj**
These attributes are better for the final state(#13628)

### CODE REVIEW

File: lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java [hunk 1]
```diff
@@ -46,8 +46,8 @@
  * uptos(position, payload). 4. start offset.
  */
 public final class Lucene99SkipWriter extends MultiLevelSkipListWriter {
-  private int[] lastSkipDoc;
-  private long[] lastSkipDocPointer;
+  private final int[] lastSkipDoc;
+  private final long[] lastSkipDocPointer;
   private long[] lastSkipPosPointer;
   private long[] lastSkipPayPointer;
```
Comment (gsmiller) [inline lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java:52]: We can make these final too if we tidy the ctor a little bit. Let's go ahead and do this if we're doing a pass on marking fields `final`?

**Review by gsmiller — COMMENTED** [review]
Always nice to mark fields `final` when possible IMO. Thanks! I noted two more we can also do this with if we just tidy the ctor a little bit to explicitly init them to `null` when not needed. Let's add that as well?

File: lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java (hunk 1 above)
Comment (gsmiller) [reply lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java:52]: @mrhbj were you planning to push an update to this PR that also marks these fields `final`? Thanks!

**Comment by mrhbj** [comment]
ok i will do this



---Original---
From: "Greg ***@***.***&gt;
Date: Fri, Aug 9, 2024 23:12 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [apache/lucene] These attributes are better for the finalstate(#13628) (PR #13630)




 
@gsmiller commented on this pull request.
 
 
In lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java:
 &gt;    private long[] lastSkipPosPointer;    private long[] lastSkipPayPointer;  
@mrhbj were you planning to push an update to this PR that also marks these fields final? Thanks!

File: lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java (hunk 1 above)
Comment (mrhbj) [reply lucene/backward-codecs/src/java/org/apache/lucene/backward_codecs/lucene99/Lucene99SkipWriter.java:52]: @gsmiller I have did that in #13639

**Comment by gsmiller** [comment]
Closing in favor of #13639

**Pull request closed without merging**